### PR TITLE
Add notifications

### LINKED ISSUE #1326: Allow notifications in the application
This will eventually allow notifications in the mobile app to keep the users engaged and in sync with the latest updates in the organization.
Proposed solution :-
- Use [firebase_messaging](https://pub.dev/packages/firebase_messaging) package to implement notifications in the frontend.
- Allow all kinds of foreground, background or terminated state notifications.
- Store FCM tokens in the database.
- Add feature in Talawa-API.

### PR DESCRIPTION
**Pull request by guptapriyanshu7**
**What kind of change does this PR introduce?**
Feature

**Issue Number:**

Fixes #1326

https://user-images.githubusercontent.com/60141300/176860639-48320887-2e9f-4499-92a6-5fef8e8662e5.mp4

**Does this PR introduce a breaking change?**
No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
@TheHazeEffect @JamaicanFriedChicken please review.

**Review by JamaicanFriedChicken — APPROVED** [review]

File: lib/utils/queries.dart [hunk 1]
```diff
@@ -68,10 +68,10 @@ class Queries {
   }
 
   //login the user
-  String loginUser(String email, String password) {
+  String loginUser(String email, String password, String? token) {
```
Comment (TheHazeEffect) [inline lib/utils/queries.dart:71]: What is this new token parameter being used for? I noticed an empty string being passed when referenced

File: lib/view_model/pre_auth_view_models/login_view_model.dart [hunk 2]
```diff
@@ -57,8 +60,8 @@ class LoginViewModel extends BaseModel {
           .pushDialog(const CustomProgressDialog(key: Key('LoginProgress')));
       databaseFunctions.init();
       try {
-        final result = await databaseFunctions
-            .gqlNonAuthMutation(queries.loginUser(email.text, password.text));
+        final result = await databaseFunctions.gqlNonAuthMutation(
+            queries.loginUser(email.text, password.text, ''));
```
Comment (TheHazeEffect) [inline lib/view_model/pre_auth_view_models/login_view_model.dart:64]: Why an empty string?

**Review by TheHazeEffect — CHANGES_REQUESTED** [review]
I was already demoed the solution so I confirmed it works 
Seems solid I just have the one query I commented on.

File: lib/view_model/pre_auth_view_models/login_view_model.dart (hunk 2 above)
Comment (guptapriyanshu7) [reply lib/view_model/pre_auth_view_models/login_view_model.dart]: Removed the empty string.

File: lib/utils/queries.dart (hunk 1 above)
Comment (guptapriyanshu7) [reply lib/utils/queries.dart]: Removed the token string.

**Review by TheHazeEffect — APPROVED** [review]
Nicely done

**Review by JamaicanFriedChicken — APPROVED** [review]

**Pull request merged**